### PR TITLE
add man to html doc

### PR DESCRIPTION
**Pull request by rgetz**
Let me know if anyone has any questions.

### CODE REVIEW

**Comment by rgetz** [comment]
For anyone who wants to have a look at the new doc, check out 

https://rgetz.github.io/libiio/v0.19/index.html

feedback welcome. As soon as this PR is closed, I will remove that personal fork.

**Comment by rgetz** [comment]
Some folks requested some tweaks, which I will do in the morning.

File: man/CMakeLists.txt [hunk 1]
```diff
@@ -11,12 +11,35 @@ if (WITH_MAN)
...
+		if (WITH_DOC)
+			file(MAKE_DIRECTORY ${CMAKE_HTML_DEST_DIR}/man1)
+			foreach(_page ${MAN1})
+				execute_process(
+					COMMAND ${MAN2HTML} -r
+					INPUT_FILE ${CMAKE_BINARY_DIR}/man/${_page}
+					OUTPUT_FILE ${CMAKE_HTML_DEST_DIR}/man1/${_page}.html
+				)
+			endforeach()
+		endif()
```
Comment (commodo) [inline man/CMakeLists.txt:42]: following a bit on comment https://github.com/analogdevicesinc/libiio/pull/413/commits/963b5e4e2b000cf0eb6a1913be1bade5be0d4e0f#r407344061  [from the formatting PR] ; maybe we should also resolve the aspect of `endif()` vs `endif__space__()`

File: CI/travis/before_install_linux [hunk 2]
```diff
@@ -14,6 +14,12 @@ install_sphinx() {
 }
 
 handle_centos() {
+	# needed for man2html and a few other popular tools
+	yum search epel-release
```
Comment (commodo) [inline CI/travis/before_install_linux:18]: curios: do we need the `yum search epel-release` command?

File: CI/travis/before_install_linux [hunk 3]
```diff
@@ -14,6 +14,12 @@ install_sphinx() {
...
+	yum info epel-release
+	yum -y install epel-release
+	yum -y update
```
Comment (commodo) [inline CI/travis/before_install_linux:21]: IIRC `yum -y update` == `apt-get update && apt-get upgrade`
i.e. it updates all installed packages;
if it's not needed [upgrade existing packages], i wouldn't do it; the time this adds to the build-run may be just a few seconds, but i'm sometimes a bit picky about build-times

for the CentOS dockers [here], it won't make much difference whether we add it or not;
they come pretty empty, so after adding this repo, we would typically install the latest versions of packages;

File: CI/travis/before_install_linux [hunk 4]
```diff
@@ -31,7 +37,7 @@ handle_centos() {
 
 	# CENTOS 6 will has issues with sphinx
 	if is_centos_at_least_ver "7" ; then
-		yum -y install doxygen
+		yum -y install doxygen man2html
 		install_sphinx
 	fi
```
Comment (commodo) [inline CI/travis/before_install_linux:42]: there are 2 `if is_centos_at_least_ver "7" ; then` blocks
merge them?

**Comment by commodo** [comment]
a few comments about the centos build-script;
but nothing major

File: CI/travis/before_install_linux (hunk 2 above)
Comment (rgetz) [reply CI/travis/before_install_linux:18]: man2html is inside the epel-release, and no where else...

File: CI/travis/before_install_linux (hunk 3 above)
Comment (rgetz) [reply CI/travis/before_install_linux:21]: It was in the all the steps I found to properly install epel-releases.

I will take it out and see if everything still all builds properly.

Comment (rgetz) [reply CI/travis/before_install_linux]: removed on latest forced push

File: CI/travis/before_install_linux (hunk 4 above)
Comment (rgetz) [reply CI/travis/before_install_linux:41]: make sense. Will fix.

File: man/CMakeLists.txt (hunk 1 above)
Comment (rgetz) [reply man/CMakeLists.txt:42]: I figured I would fix all the things I had planned, and then go back and fix the formatting issues. (which we might have to fix some style issues with codacy now, similar to the recent python fixes).

File: CI/travis/before_install_linux (hunk 4 above)
Comment (rgetz) [reply CI/travis/before_install_linux:41]: those are actually in master, I will fix in a separate commit after this is on master.

**Comment by rgetz** [comment]
Minor tweaks in the html file. (updated [preview](https://rgetz.github.io/libiio/v0.19/index.html) ) I'm just waiting on feedback from a few others, and will commit to master on Friday.

**Comment by commodo** [comment]
i see 10 commits;
is this intended?
does this need a rebase?

**Comment by rgetz** [comment]
Yeah, I will rebase against master... I'm not sure how those ended up there...

**Comment by rgetz** [comment]
should be fixed now.

**Comment by rgetz** [comment]
Ok - I think I have things rebased to master now...

**Comment by rgetz** [comment]
since I also turned on Doxygen for the C# bindings after this was approved, I will let people complain tomorrow, otherwise commit then.

File: bindings/csharp/IOBuffer.cs [hunk 5]
```diff
@@ -103,6 +103,7 @@ public void push(uint samples_count)
             circular_buffer_pushed = true;
         }
 
+	/// <summary>Submit all the samples contained in this buffer to the hardware.</summary>
```
Comment (commodo) [inline bindings/csharp/IOBuffer.cs:106]: nitpick: indentation inconsistency

**Comment by commodo** [comment]
> since I also turned on Doxygen for the C# bindings after this was approved, I will let people complain tomorrow, otherwise commit then.

one general/minor complaint i have regarding this PR, is that it reached 11 commits;
[fwiw: this complaint is one that i am periodically repeating over reviews] 

my brain can comfortably handle up-to 5 [sometimes 7 if they are small changes] commits per PR;
mostly because re-spins of 5-patches-sets are easier to re-review than 10-patches-sets;
i usually try to go over all commits [commit-by-commit], since this is the generally accepted way of doing reviews on Linux;
and i don't feel that i am doing a good-job reviewing if i don't do i like that;

for the Linux repo [on ADI Github] I ask people to try to keep them in the 5-7 commit-range; otherwise i [sometimes] defer the review to the next morning [sometimes Monday if the week is nearing to an end] to do it;
so, if they get to 7-or-more patches, this usually means 2 PRs

now, it's true that some [upstream] maintainers are fine reviewing 10-20 [sometimes 30] patches per set, and i keep wondering where they get the brain power to do it well;

anyway; it's fine to merge this [from my side] since it's mostly docs;
for code, this may have needed to sit for another 1-2 weeks [maybe more];
but some general questions would be: how good of a review do we want to do? and how do we structure it on both ends [submitter & reviewer] to do it well?

**Comment by rgetz** [comment]
@commodo : agree - for this - since it is mostly doc, I thought that the thing that would be reviewed is the actual results - https://rgetz.github.io/libiio/v0.19/index.html - not the html/css. It's hard to break things up when it is all intermingled on the resulting doc so much.

I know you already looked at the shell changes - that's why I did so many. The shell / Cmake changes are only a line or two per commit.

Will try to keep things to a reasonable number in the future.

File: bindings/csharp/IOBuffer.cs (hunk 5 above)
Comment (rgetz) [reply bindings/csharp/IOBuffer.cs]: fixed.

**Comment by rgetz** [comment]
Ok - someone noticed a spelling mistake, and then I ran the html through tidy (which I can add as part of the formatting change).

**Pull request merged**